packages/react/src/components/TimePicker/TimePicker.tsx

```diff
@@ -52,7 +51,7 @@ export const TimePicker: React.FunctionComponent<ITimePickerProps> = ({
 
   const optionsCount = getDropdownOptionsCount(increments, timeRange);
 
-  const initialValue = useConst(defaultValue || new Date());
+  const initialValue = defaultValue || new Date();
```

Comment:
I'm not sure this is the right change to make, though I'm not very familiar with the `TimePicker` code so I could be wrong 🙂

`defaultValue` is typically used for [uncontrolled React components](https://reactjs.org/docs/uncontrolled-components.html#default-values) and removing `useConst()` won't produce the correct behavior for an uncontrolled component (nor does leaving it in place).

My initial thought is that we should be using [`useControllableValue`](https://github.com/microsoft/fluentui/blob/master/packages/react-hooks/src/useControllableValue.ts) here but that may be a breaking change.

Change you provide a [Codepen example](http://aka.ms/fluentpen) of the issue as it stands today so I can understand the problem you're solving better?

Reply:
@spmonahan  I tried replicating the issue on Sandbox but it is working fine. I am facing the issue with SPFx and using TimePicker with DatePicker. First I select the date from datepicker(not today date) and then select the time. the selected date from datepicker is passed to timepicker via defaultValue property. but as soon as I select the value from timpicker it returns the todays date with the selected time. Ideally it should return the date picker date with the time selected. when I debug the control I found that initial value is set only once when the control intialized due to useConst (because at that time the defaultValue was null (as datepicker did not have value selected). If useConst is removed then the initial value will based on either the latest default value or current date. 

And you are right, useControllableValue hook should be used here.

Reply:
Sounds like you have it in hand. When you update this PR with `useControllableValue` would you also add a test cases for controlled and uncontrolled usages of `TimePicker`?

Reply:
@spmonahan I do not have expertise in writing the test cases but I tried after looking into the existing test cases and found that changed value of the control gets updated either as the value attribute of the html element or any child element but in case of TimePicker only updated/selected time gets available in the input HTML element associated with the TimePicker. tried some other ways but no luck.. may be in coming future I will be able to handle the test cases as well.

I have made the changes related to the useControllableValue hook.

Reply:
@spmonahan could you please review the changes?